src/main/scala/mesosphere/marathon/health/HealthCheck.scala

```diff
@@ -54,7 +58,13 @@ case class HealthCheck(
       path =
         if (proto.hasPath) Some(proto.getPath) else None,
       protocol = proto.getProtocol,
-      portIndex = proto.getPortIndex,
+      portIndex =
+        if (proto.hasPortIndex)
+          Some(proto.getPortIndex)
+        else if (!proto.hasPort && proto.getProtocol != Protocol.COMMAND)
```

Comment:
We are not sure. We might be for backwards compatibility. As written below, portIndex = 0 was the default int protobufs. We don't know if we can get into this situation by a combination of using an old Marathon version and migrations so we thought that we better cover it.